packages/lib/src/utils/string/index.ts

```diff
@@ -12,3 +12,12 @@ export const toSentenceCase = (string: string) => {
   }
   return string[0].toUpperCase() + string.substring(1).toLowerCase();
 };
+
+/**
+ * `toEscapedRegex` takes a string and returns an escaped version of it that can be used in a regex.
+ * @param str - The string to escape.
+ * @returns - The escaped string.
+ */
+export const toEscapedRegex = (str: string): string =>
+  // eslint-disable-next-line no-useless-escape -- This is a function to escape regex characters
+  str.replace(/[-\/\\^$*+?.()|[\]{}]/g, "\\$&");
```

Comment:
It seems like maybe this lint error is caused by malformatted regex? I think this will pass
```suggestion
export const toEscapedRegex = (str: string): string =>
    str.replace(/[-\\/\\^$*+?.()|[\]{}]/g, "\\$&");
```